model/task/task.go

```diff
@@ -2048,25 +2061,38 @@ func topologicalSort(tasks []Task) ([]Task, error) {
 	return sortedTasks, nil
 }
 
-// DeactivateTask will set the ActivatedBy field to the caller and set the active state to be false and deschedule the task
-func (t *Task) DeactivateTask(caller string) error {
-	t.ActivatedBy = caller
-	t.Activated = false
-	t.ScheduledTime = utility.ZeroTime
-
-	return DeactivateTasks([]Task{*t}, true, caller)
-}
-
 func DeactivateTasks(tasks []Task, updateDependencies bool, caller string) error {
 	taskIDs := make([]string, 0, len(tasks))
 	for _, t := range tasks {
+		// Deactivating a deactivated task is a noop.
+		if !t.Activated {
+			continue
+		}
 		if t.DisplayOnly {
 			taskIDs = append(taskIDs, t.ExecutionTasks...)
 		}
 		taskIDs = append(taskIDs, t.Id)
 	}
 
-	_, err := UpdateAll(
+	depTasksToUpdate, depTaskIDsToUpdate, err := getDependencyTasksToUpdate(taskIDs, updateDependencies)
+	if err != nil {
+		return errors.Wrap(err, "retrieving dependency tasks to deactivate")
+	}
+
+	if !evergreen.IsSystemActivator(caller) {
+		u, err := user.FindOneById(caller)
+		if err != nil {
+			return errors.Wrapf(err, "finding user '%s'", caller)
+		}
+		if u != nil {
+			settings := evergreen.GetEnvironment().Settings()
+			if err = u.CheckAndUpdateSchedulingLimit(settings, -1*(len(taskIDs)+len(depTaskIDsToUpdate))); err != nil {
```

Comment:
could you add a comment here? I think we should avoid embedded math without words

Reply:
Removed in favor of the activated param